### PR TITLE
Use the new ResponseListener from airlift

### PR DESCRIPTION
**Pull request by mbasmanova**
Also, upgrade to airlift/airbase:90 and make changes to accomodate new
stylechecks and modified Guava interfaces.

These changes compile against airlift https://github.com/airlift/airlift/pull/707

CC: @nezihyigitbasi @tdcmeehan @elonazoulay @oerling @yingsu00

### CODE REVIEW

**Comment by mbasmanova** [comment]
@elonazoulay 

> Can you add tests for the new functionality?

There is no new functionality. Just the refactoring. As a side note, I noticed that lots of tests are failing on aria-dev branch and I'm currently fixing them. I'll publish fixes as a separate PR.

**Comment by mbasmanova** [comment]
@elonazoulay 

> Looks like the build is failing with style check errors.

The build is failing because the changes depend on airlift version that's not available in Maven central. These changes are available as a PR: airlift/airlift#707

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/travis/build/prestodb/presto/presto-main/src/main/java/com/facebook/presto/operator/BufferingResponseListener.java:[18,30] cannot find symbol
  symbol:   class ResponseListener
  location: package io.airlift.http.client
[ERROR] /home/travis/build/prestodb/presto/presto-main/src/main/java/com/facebook/presto/operator/BufferingResponseListener.java:[37,20] cannot find symbol
  symbol: class ResponseListener
[ERROR] /home/travis/build/prestodb/presto/presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java:[307,68] method executeAsync in interface io.airlift.http.client.HttpClient cannot be applied to given types;
  required: io.airlift.http.client.Request,io.airlift.http.client.ResponseHandler<T,E>
  found: io.airlift.http.client.Request,com.facebook.presto.operator.HttpPageBufferClient.PageResponseHandler,com.facebook.presto.operator.BufferingResponseListener
  reason: cannot infer type-variable(s) T,E
    (actual and formal argument lists differ in length)
```

**Comment by tdcmeehan** [comment]
The build might be further failing because of the Guava update in airbase 90.

**Comment by mbasmanova** [comment]
@tdcmeehan Tim, 

> The build might be further failing because of the Guava update in airbase 90.

I'm not seeing that. `./mvnw install -DskipTests -B -T C1 -P ci` is green: https://travis-ci.org/prestodb/presto/jobs/490732196

File: presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV2.java [hunk 1]
```diff
@@ -102,7 +102,8 @@ else if (EncodingType.PATCHED_BASE.ordinal() == enc) {
             readDeltaValues(firstByte);
         }
     }
-    int trapOffsetIdx = 100000;
+
+    int trapOffsetIdx = 100_000;
```
Comment (yingsu00) [inline presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV2.java]: Do you know why this is 100_000? Maybe define a static final for it?

Comment (mbasmanova) [reply presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV2.java]: @yingsu00 I don't. This class like all other Aria readers requires substantial refactoring, but I'm not ready to do it yet.

**Comment by mbasmanova** [comment]
@tdcmeehan Tim, I ran into runtime error due upgrade to guava 0.26. Specifically, airlift.tpch is failing:

```
java.lang.NoSuchFieldError: WHITESPACE

	at io.airlift.tpch.DistributionLoader.loadDistributions(DistributionLoader.java:108)
	at io.airlift.tpch.DistributionLoader.loadDistribution(DistributionLoader.java:47)
	at io.airlift.tpch.Distributions.getDefaultDistributions(Distributions.java:38)
	at com.facebook.presto.tpch.TpchMetadata.<clinit>(TpchMetadata.java:104)
	at com.facebook.presto.tests.H2QueryRunner.<init>(H2QueryRunner.java:96)
	at com.facebook.presto.tests.AbstractTestQueryFramework.init(AbstractTestQueryFramework.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:169)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
```

**Pull request closed without merging**